handlers/omni/v2/fileformat/csv/reader.go

```diff
@@ -0,0 +1,156 @@
+package omniv2csv
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/antchfx/xpath"
+	"github.com/jf-tech/go-corelib/caches"
+	"github.com/jf-tech/go-corelib/ios"
+	"github.com/jf-tech/go-corelib/maths"
+
+	"github.com/jf-tech/omniparser/idr"
+)
+
+// ErrInvalidHeader indicates the header of the CSV input is corrupted, mismatched, or simply
+// unreadable. This is a fatal, non-continuable error.
+type ErrInvalidHeader string
+
+func (e ErrInvalidHeader) Error() string { return string(e) }
+
+// IsErrInvalidHeader checks if an err is of ErrInvalidHeader type.
+func IsErrInvalidHeader(err error) bool {
+	switch err.(type) {
+	case ErrInvalidHeader:
+		return true
+	default:
+		return false
+	}
+}
+
+type reader struct {
+	inputName     string
+	decl          *fileDecl
+	xpath         *xpath.Expr
```

Comment:
Do we need to explain here the meaning of xpath in csv reader?

Reply:
No. This is internal stuff. Just like similarly we didn't explain why xpath is used in json reader. We will have usage doc and development doc later explaining the idr.Node + xpath based parsing and transform.